### DISCUSSION TITLE
[solved] tox4: unicode related regression

### DISCUSSION THREAD

**Discussion by ssbarnea** [tox-dev/tox#1803] [General]
I suspect that tox4 does something different when it comes to passenv because I found a very weird bug: `tox -e py36` passed as expected but `tox4 -e py36` fails.

So far, this happens only with **py36** and only on **macos** on https://github.com/sphinx-contrib/sphinxcontrib-programoutput/ project.

I will make PR to try to reproduce it on CI.

The first thing that went into my mind was that maybe LANG, LC_ALL or LC_CTYPE were not passed, but is not that.

Update: is not macos specific but it it seems related to py36, sometimes. See https://github.com/sphinx-contrib/sphinxcontrib-programoutput/pull/22

**Comment by gaborbernat** [comment]
Did you try with the latest commit on the rewrite, https://github.com/tox-dev/tox/commit/05c23cb44622024c39cc6e317fa75388aaa3c845 should fix that.

**Reply by gaborbernat** [reply]
A reported issue for this https://github.com/tox-dev/tox/issues/1781

**Reply by ssbarnea** [reply]
Thanks. I will wait for the next pre-release and start testing it again. I have some really heavy/complex tox.ini files and likely that when it will work fine on all it would be ready for release.

I only hope you do not burn-down due to the upcoming feedback.

**Reply by gaborbernat** [reply]
The next release probably not happening, at least until the end of the week. Have some other big changes needed, now working on addressing https://github.com/tox-dev/tox/issues/1782, which requires a redesign of how we do packaging a bit. But getting close to nailing it down.

> I only hope you do not burn-down due to the upcoming feedback.

Thanks, as it was with virtualenv too, it's a marathon, not a sprint. tox 4 to be stable with plugins likely not going to happen before the end of the year. However, by April, we should have a first rough working version we can release as ``4.0.0``. Just inching there, step by step, one issue at a time. It's important that people try the alpha/beta releases so we can detect major flaws in my approach/design before we go public. So your effort on this is much appreciated!

**Reply by ssbarnea** [reply]
Usually I would have started to propose patches on found bugs but being realistic, I am already overloaded with doing maintenance on [other packages](https://pypi.org/user/ssbarnea/).

**Reply by gaborbernat** [reply]
4.0.0a3 is out and should fix this, https://tox.readthedocs.io/en/rewrite/changelog.html#v4-0-0a3-2021-01-13
